### PR TITLE
f-fix/gh-344-refresh-data-on-new-request

### PR DESCRIPTION
**Pull request by aberonja**
Fixed team calendar refresh after request create.

GH issue: #344

### CODE REVIEW

File: src/main/frontend/src/store/actions/yearActions.js [hunk 1]
```diff
@@ -32,5 +32,5 @@ export const createYear = createAction(CREATE_YEAR);
 export const createYearSuccessful = createAction(CREATE_YEAR_SUCCESSFUL);
 export const createYearError = createAction(CREATE_YEAR_ERROR);
 
-export const SET_SELECTED_YEAR = 'SET_SELECTED_YEAR';
-export const setSelectedYear = createAction(SET_SELECTED_YEAR);
+export const SET_SELECTED_DATE = 'SET_SELECTED_DATE';
```
Comment (BraneOpacic21) [inline src/main/frontend/src/store/actions/yearActions.js]: Just a thought: maybe decouple actions and where they are created in separate places? 🥉

Comment (aberonja) [reply src/main/frontend/src/store/actions/yearActions.js]: I like it, and certainly would have it in mind for future refactoring, but at the moment there is a lot of files that would need to be refactored. Also, I'll leave a comment not to forget it.

File: src/main/frontend/src/store/sagas/requestsSaga.js [hunk 2]
```diff
@@ -87,10 +122,10 @@ const handleRequestSaga = function*(action) {
...
     yield put(requestRequestsForApproval());
   } else {
-    const selectedYear = yield select(getSelectedYear);
-    yield put(requestUserDays({ year: selectedYear }));
-    yield put(requestAllowance(selectedYear));
-    yield put(requestLeaveRequestsForYear(selectedYear));
+    const selectedDate = yield select(getSelectedDate);
+    yield put(requestUserDays(selectedDate));
+    yield put(requestAllowance(selectedDate));
+    yield put(requestLeaveRequestsForYear(selectedDate));
```
Comment (BraneOpacic21) [inline src/main/frontend/src/store/sagas/requestsSaga.js]: Maybe refactor this to remove **_if else_** block with:

`isApprover?  requestsApprovalSaga() : requestsSaga()`

.. just with better saga names 😃

File: src/main/frontend/src/store/sagas/requestsSaga.js [hunk 3]
```diff
@@ -48,31 +50,64 @@ const createLeaveRequestSaga = function*(action) {
...
+      requestTeamRequestsForMonth({
+        teamId: user.teamId,
+        date: moment(selectedDate, 'YYYY-MMM').format('YYYY-MM')
+      })
+    );
+  } else {
+    yield put(requestUserDays(selectedDate));
+    yield put(requestAllowance(selectedDate));
+    yield put(requestLeaveRequestsForYear(selectedDate));
+  }
```
Comment (BraneOpacic21) [inline src/main/frontend/src/store/sagas/requestsSaga.js]: As far as I can see this part (13 lines above from here) is repeated three times.

Refactor all of this to be called as a function maybe?

File: src/main/frontend/src/components/team-calendar/MonthHandlerContainer.js [hunk 4]
```diff
@@ -74,6 +76,11 @@ class TeamCalendarContainer extends Component {
         this.state.monthsWithAllowance.indexOf(selectedMonth)
       ]
     });
+    this.props.setSelectedDate(
+      this.state.monthsWithAllowance[
+        this.state.monthsWithAllowance.indexOf(selectedMonth)
```
Comment (BraneOpacic21) [inline src/main/frontend/src/components/team-calendar/MonthHandlerContainer.js]: Double-check this, seems off.

File: src/main/frontend/src/store/actions/yearActions.js (hunk 1 above)
Comment (BraneOpacic21) [reply src/main/frontend/src/store/actions/yearActions.js]: 👍

File: src/main/frontend/src/store/sagas/requestsSaga.js (hunk 2 above)
Comment (aberonja) [reply src/main/frontend/src/store/sagas/requestsSaga.js]: @iscekic?

**Pull request merged**